cpp/tests/experimental/spatial/point_polygon_distance_test.cu

```diff
@@ -0,0 +1,427 @@
+                     {0.0, 0.17677669529663687});
+}
+
+TYPED_TEST(PairwisePointPolygonDistanceTest, ThreePolygons)
+{
+  using T = TypeParam;
+  using P = vec_2d<T>;
+
+  CUSPATIAL_RUN_TEST(this->run_single,
+                     {{P{1, 1}}, {P{2, 2}}, {P{1.5, 0}}},
+                     {0, 1, 2, 3},
+                     {0, 2, 3, 6},
+                     {0, 4, 8, 12, 17, 22, 27},
+                     {// POLYGON ((0 1, -1 -1, 1 -1, 0 1), (0 0.5, 0.5 -0.5, -0.5 -0.5, 0 0.5))
+                      P{0, 1},
+                      P{-1, -1},
+                      P{1, -1},
+                      P{0, 1},
+                      P{0, 0.5},
+                      P{0.5, -0.5},
+                      P{-0.5, -0.5},
+                      P{0, 0.5},
+                      // POLYGON ((1 1, 1 2, 2 1, 1 1))
+                      P{1, 1},
+                      P{1, 2},
+                      P{2, 1},
+                      P{1, 1},
+                      //  POLYGON (
+                      //      (-3 -3, 3 -3, 3 3, -3 3, -3 -3),
+                      //      (-2 -2, -1 -2, -1 2, -2 2, -2 -2),
+                      //      (2 2, 2 -2, 1 -2, 1 2, 2 2)
+                      //  )
+                      P{-3, -3},
+                      P{3, -3},
+                      P{3, 3},
+                      P{-3, 3},
+                      P{-3, -3},
+                      P{-2, -2},
+                      P{-1, -2},
+                      P{-1, 2},
+                      P{-2, 2},
+                      P{-2, -2},
+                      P{2, 2},
+                      P{2, -2},
+                      P{1, -2},
+                      P{1, 2},
+                      P{2, 2}},
+                     {0.894427190999916, 0.7071067811865476, 0.5});
+}
+
+TYPED_TEST(PairwisePointPolygonDistanceTest, OnePairMultiPointOnePolygon)
+{
+  using T = TypeParam;
+  using P = vec_2d<T>;
+
+  CUSPATIAL_RUN_TEST(this->run_single,
+                     {{P{0, 3}, P{2, 0}}},
+                     {0, 1},
+                     {0, 1},
+                     {0, 5},
+                     {P{0, 1}, P{-1, -1}, P{1, -1}, P{0, 1}},
+                     {1.3416407864998738});
+}
+
+TYPED_TEST(PairwisePointPolygonDistanceTest, OnePairMultiPointOnePolygon2)
+{
+  using T = TypeParam;
+  using P = vec_2d<T>;
+
+  CUSPATIAL_RUN_TEST(this->run_single,
+                     {{P{0, 3}, P{0, 0}}},
+                     {0, 1},
+                     {0, 1},
+                     {0, 5},
+                     {P{0, 1}, P{-1, -1}, P{1, -1}, P{0, 1}},
+                     {0.0});
+}
+
+TYPED_TEST(PairwisePointPolygonDistanceTest, TwoPairMultiPointOnePolygon2)
+{
+  using T = TypeParam;
+  using P = vec_2d<T>;
+
+  CUSPATIAL_RUN_TEST(
+    this->run_single,
+    {{P{0, 2}, P{0, 0}}, {P{1, 1}, P{-1, -1}}},
+    {0, 1, 2},
+    {0, 1, 2},
+    {0, 5, 9},
+    {P{-1, -1}, P{1, -1}, P{1, 1}, P{-1, 1}, P{-1, -1}, P{-1, 1}, P{1, 1}, P{0, -1}, P{-1, 1}},
+    {0.0, 0.0});
+}
```

Comment:
Do you have any hole tests in here? Are holes currently supported?

Consider a square polygon of radius 2 containing an interior ring of radius 1 and a point in the middle. Is the distance of the point from the polygon 0.5?

Reply:
Holes are supported, see test: 
- `OnePairOnePolygonTwoRings`
- `OnePairOnePolygonTwoRings2`
- `OnePairOnePolygonTwoRings3`

Test case is a box with box hole inside. The three tests are 3 points in the hole, in the polygon area and outside the polygon.

I will add comments to the tests to make them easier to read.